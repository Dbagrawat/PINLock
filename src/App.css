/*
app variables
*/

:root {
    /* font */
    --main-font: "Orbitron", sans-serif;
}

/*
app CSS reset
*/

body,
div,
p {
    margin: 0;
    padding: 0;
}

.pinlock-wrapper {
    width: 300px;
    height: 400px;
    border: 1px solid #ccc;
    border-radius: 10px;
    margin: 100px auto;
    /*  */
}

h1,
h2 {
    text-align: center;
}
h3 {
    margin: 0px;
}
.code {
    display: block;
    margin-top: 20px;
    padding: 0;
    width: 50px;
    margin-left: auto;
    margin-right: auto;
}
.code li {
    list-style: none;
    display: inline-block;
    margin-left: 5px;
    width: 5px;
    height: 5px;
    border: 1px solid #ccc;
    border-radius: 5px;
    transition: background 0.5s ease-in-out;
}
.keypad {
    width: 100%;
    height: 100%;
    padding: 0;
    margin: 20px;
}
.dial {
    font-family: Helvetica;
    text-align: center;
    vertical-align: middle;
    line-height: 55px;
    font-size: 20px;
    color: black;
    list-style: none;
    display: inline-block;
    width: 75px;
    height: 50px;
    margin: 5px;
    border: 1px solid #ccc;
    border-radius: 1%;
    background: white;
    position: relative;
}
.dial:before {
    border-radius: 1%;
    content: "";
    /* background-image: linear-gradient(to bottom, #4bcd7d 0%, #1ca844 100%); */
    padding: 2px;
    width: 75px;
    height: 50px;
    top: -2px;
    left: -2px;
    position: absolute;
    z-index: -1;
}

.middle {
    margin-left: 95px;
    margin-right: 90px;
}

.dial:active {
    background: #7d7d7d;
}
.fill {
    background: #000;
}
.passcode {
    text-align: center;
}
.passcode input {
    border: none;
    text-align: center;
}
