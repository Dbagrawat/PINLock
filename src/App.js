import React, { useState } from "react";
import "./App.css";
import Keypad from "./components/keyPad";
import Passcode from "./components/passcode";
import { properties } from "./properties";

function App() {
    const [unlock, setUnlcok] = useState(false);
    const [errorCount, setErrorCount] = useState([]);
    const [passcode, setPasscode] = useState([]);

    const handleClear = () => {
        setUnlcok(0);
        setPasscode([]);
        setErrorCount([]);
    };

    const handleUnlock = () => {
        let pass = passcode.join("");
        if (pass === properties.correctCombination) {
            setUnlcok(true);
            setPasscode([]);
        } else {
            setPasscode([]);
            setErrorCount((prevCount) => prevCount + 1);
        }
    };

    document.onkeypress = function (e) {
        e = e || window.event;
        var charCode = e.charCode || e.keyCode,
            character = String.fromCharCode(charCode);
        setPasscode([...passcode, character]);
        if (charCode === 13) handleUnlock();
    };

    const handleKeypad = (e) => {
        let keyPadValue = e.target.innerHTML;
        if (keyPadValue === properties.clear) {
            handleClear();
        } else if (keyPadValue === properties.unlock) {
            handleUnlock();
        } else {
            setPasscode([...passcode, keyPadValue]);
        }
    };

    return (
        <div className="pinlock-wrapper">
            <h2>{unlock ? properties.unlocked : properties.locked}</h2>
            <Passcode
                passcode={passcode.join("")}
                errorCount={errorCount.length}
            />
            <div>
                {unlock || errorCount.length === 3 ? (
                    ""
                ) : (
                    <Keypad handleKeypad={handleKeypad} />
                )}
            </div>
        </div>
    );
}

export default App;
