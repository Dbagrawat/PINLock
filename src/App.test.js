import React from "react";
import Enzyme, { shallow, render, mount } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import toJson from "enzyme-to-json";
import App from "./App";
import Keypad from "./components/keyPad";
import Passcode from "./components/passcode";
import DisplayMessage from "./components/displayMessage";

Enzyme.configure({ adapter: new Adapter() });

it("App component renders correctly ", () => {
    const wrapper = shallow(<App />);
    expect(toJson(wrapper)).toMatchSnapshot();
});

it("KeyPad component renders correctly ", () => {
    const wrapper = shallow(<Keypad />);
    expect(toJson(wrapper)).toMatchSnapshot();
});

it("Passcode component renders correctly ", () => {
    const wrapper = shallow(<Passcode />);
    expect(toJson(wrapper)).toMatchSnapshot();
});

it("Message component renders correctly ", () => {
    const wrapper = shallow(<DisplayMessage />);
    expect(toJson(wrapper)).toMatchSnapshot();
});

it("should render a password input", () => {
    const wrapper = shallow(<Passcode />);
    expect(wrapper.find('input[name="password"]').exists()).toBe(true);
});
