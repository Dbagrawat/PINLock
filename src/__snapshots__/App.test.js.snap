// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders correctly  1`] = `
<div
  className="pinlock-wrapper"
>
  <h2>
    Locked
  </h2>
  <passcode
    errorCount={0}
    passcode=""
  />
  <div>
    <keyPad
      handleKeypad={[Function]}
    />
  </div>
</div>
`;

exports[`KeyPad component renders correctly  1`] = `
<ul
  className="keypad"
>
  <li
    className="dial"
    key="0"
    value="1"
  >
    1
  </li>
  <li
    className="dial"
    key="1"
    value="2"
  >
    2
  </li>
  <li
    className="dial"
    key="2"
    value="3"
  >
    3
  </li>
  <li
    className="dial"
    key="3"
    value="4"
  >
    4
  </li>
  <li
    className="dial"
    key="4"
    value="5"
  >
    5
  </li>
  <li
    className="dial"
    key="5"
    value="6"
  >
    6
  </li>
  <li
    className="dial"
    key="6"
    value="7"
  >
    7
  </li>
  <li
    className="dial"
    key="7"
    value="8"
  >
    8
  </li>
  <li
    className="dial"
    key="8"
    value="9"
  >
    9
  </li>
  <li
    className="dial"
    key="9"
    value="Clear"
  >
    Clear
  </li>
  <li
    className="dial"
    key="10"
    value="0"
  >
    0
  </li>
  <li
    className="dial"
    key="11"
    value="Unlock"
  >
    Unlock
  </li>
</ul>
`;

exports[`Message component renders correctly  1`] = `
<Fragment>
   
</Fragment>
`;

exports[`Passcode component renders correctly  1`] = `
<div
  className="passcode"
>
  <input
    className="passcode"
    name="password"
    readOnly={true}
    type="password"
  />
  <h3>
    <displayMessage
      message={
        Object {
          "attempt1": 1,
          "attempt2": 2,
          "attempt3": 3,
          "clear": "Clear",
          "correctCombination": "1234",
          "incorrectentry": "Incorrect entry",
          "locked": "Locked",
          "toomanyattempts": "Too many incorrect attempts",
          "unlock": "Unlock",
          "unlocked": "Unlocked",
        }
      }
    />
  </h3>
</div>
`;
